migration/migration.go

```diff
@@ -82,3 +85,48 @@ func revision0000(txn db.Transaction) error {
 	}
 	return it.Close()
 }
+
+// relocateContractStorageRootKeys moves contract root keys to new locations and deletes the previous locations.
+//
+// Before: the key to the root in a contract storage trie was stored at db.ContractRootKey+<contractAddress>.
+// After: the key to the root of the contract storage trie is stored at db.ContractStorage+<contractAddress>.
+//
+// This enables us to remove the db.ContractRootKey prefix.
+func relocateContractStorageRootKeys(txn db.Transaction) error {
+	it, err := txn.NewIterator()
+	if err != nil {
+		return err
+	}
+
+	// Iterate over all entries in the old bucket, moving each one to its new location.
+	oldPrefix := db.ContractRootKey.Key()
+	it.SetOptions(&pebble.IterOptions{
+		LowerBound: oldPrefix,
+		UpperBound: (db.ContractRootKey + 1).Key(), // The prefix that succeeds db.ContractRootKey.
+	})
+	for it.First(); it.Valid(); it.Next() {
+		oldKey := it.Key()
```

Comment:
How about something like;

https://github.com/NethermindEth/juno/blob/b969344250d71aebe8b54ce72fd993ab1fb9c694/blockchain/blockchain.go#L402-L406

that way, we dont need to introduce `SetOptions` or `First`

Reply:
Good idea.